### DISCUSSION TITLE
Unique constrains if deleted

### DISCUSSION THREAD

**Discussion by ggukggukee** [prisma/prisma#17886] [Q&A]
Hello! I have schema like this: 

```prisma
model UserCourse {
  id Int @id @default(autoincrement())
  user User @relation(fields: [userId], references: [id])
  userId Int
  course Course @relation(fields: [courseId], references: [id])
  courseId Int
  isDeleted Boolean @default(false)
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  lectures Lecture[]
  @@unique([userId, courseId, isDeleted])
}
```
That means that I there will be no same user-course that not deleted and no same user-course that is deleted.
How can I rewrite my schema that I can have multiple deleted user-course but have only one not deleted?

**Comment by arosemena** [comment]
You can do it with a unique index

```sql
CREATE UNIQUE INDEX unique_user_course_not_deleted
    on "UserCourse" ("userId", "courseId") where "isDeleted" = false;
```

To apply it create an empty migration with `npx prisma migrate dev --create-only` and use that sql

**Reply by ggukggukee** [reply]
How can I implement it in prisma.schema?

**Reply by arosemena** [reply]
I'm not seeing any way to do this custom unique constraint with the prisma schema itself at this time, when you have custom functionality you have to do it through a custom migration.

**Reply by nikolasburk** [reply]
Thanks @arosemena for your help here 🎉 @ggukggukee, I'm just dropping in to share the docs for customizing migrations which should help you with this use case: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations

**Reply by ggukggukee** [reply]
if I will use custom migration will it fire P2002 error for duplicates?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
